src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -692,14 +693,15 @@ private bool TryGetSpanConversion(TypeSymbol type, out MethodSymbol? asSpanMetho
                 return false;
             }
 
-            if (namedType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_Span_T), TypeCompareKind.ConsiderEverything)
-                || namedType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.ConsiderEverything))
+            if ((!writableOnly && namedType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.ConsiderEverything))
+                || namedType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_Span_T), TypeCompareKind.ConsiderEverything))
             {
                 asSpanMethod = null;
                 return true;
             }
 
-            if (namedType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_Collections_Immutable_ImmutableArray_T), TypeCompareKind.ConsiderEverything)
+            if (!writableOnly
```

Comment:
>!writableOnly [](http://example.com/codeflow?start=16&length=13)

It looks like we are returning conversion to a writable ```Span``` regardless. Why do the check then? And is there a conversion to ```ReadOnlySpan``` we can use?

Reply:
It looks like ```ImmutableArray.AsSpan``` returns ```ReadOnlySpan ```. It all makes sense now.

Reply:
[ImmutableArray<T>.AsSpan](https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutablearray-1.asspan?view=net-8.0#system-collections-immutable-immutablearray-1-asspan) returns `ReadOnlySpan`, so in this path, we are returning a conversion to `ReadOnlySpan`. Perhaps it would be helpful to add a comment about the signature of this method.